server/src/api/customers.ts

```diff
@@ -39,4 +40,22 @@ customerRouter.get(
   }
 );
 
+/**
+ * Create a customer if they do not already exist.
+ */
+customerRouter.post(
+  '/',
+  async (req: Request, res: Response, next: NextFunction) => {
+    const customerData: CustomerPayload = req.body;
+
+    try {
+      const customer = await customerService.addCustomer(customerData);
+      res.status(201).send(customer);
```

Comment:
You should also include 'Location' header in POST response.

> If a resource has been created on the origin server, the response SHOULD be 201 (Created) and contain an entity which describes the status of the request and refers to the new resource, and a Location header (see section 14.30).

from [here](https://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html).

Also, since you are returning the inserted object in the response body, should also include 'Content-Location' header, as mentioned in [this comment](https://stackoverflow.com/a/1832046) and the [definition of the header](https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.14).

Reply:
@karenfrilya97 ok :D